### ISSUE TITLE
Security Bug: Unchecked call to gcry_control

### ISSUE THREAD

**Issue by yujokang** [FreeRDP/Remmina#830]
Dear Remmina developers,

We are security researchers at Columbia University and the University of Virginia. As part of a research project, we have built a tool for automatically finding error handling bugs and are testing it on various cryptographic libraries and applications that use them.

In the source code at the time of commit 5130ca586a7823fddba1dfea0b204c9c46448f02, we discovered that GCrypt's gcry_control is sometimes called without checking its return value for an error, which may cause the program to be initialized without the desired, secure settings.

Please let us know how you intend to address the following issue:

When gcry_control is called with GCRYCTL_SET_THREAD_CBS, it can return a non-zero error value for reasons beyond the control of the caller, so it needs to be checked; otherwise the program may not be properly initialized. In particular, we found such use in:

remmina/src/remmina.c, line 283:

``` C
int main(void)
{
...
 gcry_control (GCRYCTL_SET_THREAD_CBS, &gcry_threads_pthread);
...
}
```

Thank you,
Yuan Kang
##

**Comment by antenore** [comment]
Thanks so much for this report.

We will take a look ASAP and let you know.

@giox069

**Comment by antenore** [comment]
I'm working on it

https://github.com/FreeRDP/Remmina/commit/bfdc406098f58744a8b20adbddd1eeac7242529e

**Comment by antenore** [comment]
closed via 01453a374
